Backend/Microservices/ChatService/src/Chat.Data/Services/CacheService.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Chat.Application.DTOs;
+using Chat.Application.Interfaces;
+using Chat.Domain.Entities;
+using Microsoft.Extensions.Caching.Distributed;
+using System.Text.Json;
+
+namespace Chat.Data.Services;
+
+public class CacheService : ICacheService
+{
+    private readonly IUserRepository _userRepository;
+    private readonly IMessageRepository _messageRepository;
+    private readonly IDistributedCache _cache;
+
+    public CacheService(IUserRepository userRepository, IMessageRepository messageRepository, IDistributedCache cache)
+    {
+        _userRepository = userRepository;
+        _messageRepository = messageRepository;
+        _cache = cache;
+    }
```

Comment:
Mb inject IUnitOfWotk to constructor to get repositories? You use both a unit of work and separate repositories. I think, you need either inject repositories as IUnitOfWork fields in Program.cs or inject IUnitOfWork in this class